packages/cli/src/utils/directory.test.ts

```diff
@@ -8,77 +8,210 @@ const pathsToMatch = (expected: string, desired: string) => {
+      const { userStoreConfigFile: userStoreConfigFileWithBase } = withBasePath(basePath)
+
+      expect(pathsToMatch(userStoreConfigFileWithBase, './faststore.config.js')).toBe(true)
     })
   })
 
   describe('tmpStoreConfigFile', () => {
     it('returns the path of the faststore.config file in the customizations dir', () => {
-      expect(pathsToMatch(tmpStoreConfigFile, './.faststore/src/customizations/faststore.config.js')).toBe(true)
+      const { tmpStoreConfigFile: tmpStoreConfigFileWithBase } = withBasePath(basePath)
+
+      expect(pathsToMatch(tmpStoreConfigFileWithBase, './.faststore/src/customizations/faststore.config.js')).toBe(true)
     })
   })
 })
 
+describe('withBasePath as an arbitrary dir', () => {
+  const basePath = path.join(__dirname, '..', '__mocks__', 'store')
+
+  describe('coreDir', () => {
+    it('is the faststoreDir + core', () => {
+      const { coreDir: coreDirWithBase } = withBasePath(basePath)
+
+      expect(pathsToMatch(coreDirWithBase, './src/__mocks__/store/node_modules/@faststore/core')).toBe(true)
+    })
+
+    describe('when is in a monorepo', () => {
+      it.todo('can look at its parent until it reaches the monorepo directory')
+    })
+  })
+
+  describe('tmpDir', () => {
+    it('is the basePath + .faststore', () => {
+      const { tmpDir: tmpDirWithBase } = withBasePath(basePath)
+
+      expect(pathsToMatch(tmpDirWithBase, './src/__mocks__/store/.faststore')).toBe(true)
+    })
+  })
+
+  describe('userDir', () => {
+    it("returns the directory of the starter's package.json", () => {
+      const { userSrcDir: userSrcDirWithBase } = withBasePath(basePath)
+
+      expect(pathsToMatch(userSrcDirWithBase, './src/__mocks__/store/src')).toBe(true)
+    })
+  })
+
+  describe('tmpCustomizationsSrcDir', () => {
+    it('returns the directory of customizations directory on the tmp dir', () => {
+      const { tmpCustomizationsSrcDir: tmpCustomizationsSrcDirWithBase } = withBasePath(basePath)
+
+      expect(pathsToMatch(tmpCustomizationsSrcDirWithBase, './src/__mocks__/store/.faststore/src/customizations/src')).toBe(true)
+    })
+  })
+
+  describe('userThemesFileDir', () => {
+    it("returns the directory of the starter's theme directory", () => {
+      const { userThemesFileDir: userThemesFileDirWithBase } = withBasePath(basePath)
+
+      expect(pathsToMatch(userThemesFileDirWithBase, './src/__mocks__/store/src/themes')).toBe(true)
+    })
+  })
+
+  describe('tmpThemesCustomizationsFile', () => {
+    it('returns the path of the theme file on the .faststore dir', () => {
+      const { tmpThemesCustomizationsFile: tmpThemesCustomizationsFileWithBase } = withBasePath(basePath)
+
+      expect(pathsToMatch(tmpThemesCustomizationsFileWithBase, './src/__mocks__/store/.faststore/src/customizations/src/themes/index.scss')).toBe(true)
+    })
+  })
+
+  describe('tmpCMSDir', () => {
+    it('returns the path of the theme file on the .faststore dir', () => {
```

Comment:
```suggestion
    it('returns the path of the CMS dir on the .faststore dir', () => {
```

Reply:
@gvc I believe you missed this comment 👆

Reply:
Oops, sorry! Fixed it now!